packages/backend/src/database/migrations/20240109124309_project_group_access.ts

```diff
@@ -0,0 +1,40 @@
+import { Knex } from 'knex';
+
+const PROJECTS_TABLE_NAME = 'projects';
+const GROUPS_TABLE_NAME = 'groups';
+const PROJECT_MEMBERSHIP_ROLES_TABLE_NAME = 'project_membership_roles';
+const PROJECT_GROUP_ACCESS_TABLE_NAME = 'project_group_access';
+
+export async function up(knex: Knex): Promise<void> {
+    if (!(await knex.schema.hasTable(PROJECT_GROUP_ACCESS_TABLE_NAME))) {
+        await knex.schema.createTable(
+            PROJECT_GROUP_ACCESS_TABLE_NAME,
+            (table) => {
+                table
+                    .integer('project_id')
```

Comment:
We used ids in the past for foreign keys but new tables use uuids to simplify the queries in models.

Reply:
With this change we can stop using knex.raw in favour or other functions.

```
const query = this.database.raw(
            `
            DELETE FROM ${ProjectGroupAccessTableName}
            WHERE project_id = (
                SELECT project_id
                FROM ${ProjectTableName}
                WHERE project_uuid = :projectUuid
            )
            AND group_uuid = :groupUuid
            RETURNING *
            `,
            { projectUuid, groupUuid },
        );
```
With 
```
const query = this.database.delete().where('project_uuid', projectUuid);
```

And 

```
     const query = this.database.raw<{
            rows: DBProjectGroupAccess[];
        }>(
            `
            INSERT INTO ${ProjectGroupAccessTableName} (project_id, group_uuid, role)
            SELECT ${ProjectTableName}.project_id, ${GroupTableName}.group_uuid, :role
            FROM ${ProjectTableName}
            INNER JOIN ${GroupTableName} ON ${ProjectTableName}.organization_id = ${GroupTableName}.organization_id
            WHERE ${ProjectTableName}.project_uuid = :projectUuid
            AND ${GroupTableName}.group_uuid = :groupUuid
            AND ${ProjectTableName}.organization_id = ${GroupTableName}.organization_id
            ON CONFLICT DO NOTHING
            RETURNING *
            `,
            { projectUuid, groupUuid, role },
        )
```

With 

```
   const query = this.database..insert( { projectUuid, groupUuid, role })
```

Reply:
@ZeRego yeah, that makes sense! I'll make the changes 🙌 

```
const query = this.database.insert( { projectUuid, groupUuid, role })
```

this is simple but lacks db integrity check where group and project should belong to the same org. I'll try the KNEX way of that query